crates/aptos/src/governance/mod.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{
+    AccountAddressWrapper, CliError, CliTypedResult, PromptOptions, TransactionOptions,
+};
+use crate::common::utils::prompt_yes_with_override;
+use crate::{CliCommand, CliResult};
+use aptos_crypto::HashValue;
+use aptos_rest_client::Transaction;
+use aptos_types::account_address::AccountAddress;
+use async_trait::async_trait;
+use clap::Parser;
+use reqwest::Url;
+use serde::Deserialize;
+use serde::Serialize;
+use std::fmt::Formatter;
+
+/// Tool for on-chain governance
+///
+#[derive(Parser)]
+pub enum GovernanceTool {
+    Propose(SubmitProposal),
+    Vote(SubmitVote),
+}
+
+impl GovernanceTool {
+    pub async fn execute(self) -> CliResult {
+        use GovernanceTool::*;
+        match self {
+            Propose(tool) => tool.execute_serialized().await,
+            Vote(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Submit proposal to other validators to be proposed on
+#[derive(Parser)]
+pub struct SubmitProposal {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Delegated pool address to submit proposal on behalf of
+    #[clap(long)]
+    pub(crate) pool_address: AccountAddressWrapper,
+    /// Execution hash of the script to be voted on
+    #[clap(long, parse(try_from_str = read_hex_hash))]
+    pub(crate) execution_hash: HashValue,
```

Comment:
slightly tangentially; have we ever told people what it should be? (assuming hex encoded sha256; but we could use b64 for brevity)